packages/lib/services/rest/routes/notes.ts

```diff
@@ -407,6 +414,10 @@ export default async function(request: Request, id: string = null, link: string
 
 		const newProps = request.bodyJson(readonlyProperties('PUT'));
 		if (!('user_updated_time' in newProps)) newProps.user_updated_time = timestamp;
+		if (newProps.is_todo === 0) {
+			newProps.todo_due = 0;
+			newProps.todo_completed = 0;
+		}
```

Comment:
I honestly don't know if these props are required to be 0 in this case. Is that how the app works? If so, where's the logic for it? If it's somewhere in the UI, then it should be refactored and moved to the model, then reused here. I really don't want to have these pieces of business logic here and there, it's too hard to maintain and to keep it consistent.

Reply:
It more or less what the app does. `changeNoteType` in the Note model sets these fields when changing the type. But that method works because you can't change the note type and one of the other two fields at exactly the same time. With the api you can, so without this check you can have a note with a todo date. 

![image](https://user-images.githubusercontent.com/10060747/204368382-64ff69a4-cfdf-4ed4-997c-3d0d98e4e85e.png)

If that's not a problem (or the responsibility of whoever calls the api) these checks can be removed. Otherwise I'll see if I can move them to a location that makes more sense.